deeprankcore/trainer.py

```diff
@@ -15,76 +15,87 @@
+                transform_sigmoid: Optional[bool] = False,
+                metrics_exporters: Optional[List[MetricsExporter]] = None,
+                metrics_output_dir: str = './metrics'
+            ):
         """Class from which the network is trained, evaluated and tested
 
         Args:
-            dataset_train (HDF5DataSet object, required): training set used during training.
-                Can't be None if pretrained_model is also None. Defaults to None.
-            dataset_val (HDF5DataSet object, optional): evaluation set used during training.
-                Defaults to None. If None, training set will be split randomly into training set and
-                validation set during training, using val_size parameter
-            dataset_test (HDF5DataSet object, optional): independent evaluation set. Defaults to None.
-            Net (function, required): neural network class (ex. GINet, Foutnet etc.).
+            neuralnet (function, required): neural network class (ex. GINet, Foutnet etc.).
                 It should subclass torch.nn.Module, and it shouldn't be specific to regression or classification
                 in terms of output shape (Trainer class takes care of formatting the output shape according to the task).
                 More specifically, in classification task cases, softmax shouldn't be used as the last activation function.
+
+            dataset_train (GraphDataset object, required): training set used during training.
+                Can't be None if pretrained_model is also None. Defaults to None.
+
+            dataset_val (GraphDataset object, optional): evaluation set used during training.
+                Defaults to None. If None, training set will be split randomly into training set and
+                validation set during training, using val_size parameter
+
+            dataset_test (GraphDataset object, optional): independent evaluation set. Defaults to None.
+
             val_size (float or int, optional): fraction of dataset (if float) or number of datapoints (if int)
                 to use for validation.
                 - Should be set to 0 if no validation set is needed.
                 - Should be not set (None) if dataset_val is not None.
-                Defaults to None, and it is set to 0.25 in _DivideDataSet function if no dataset_val is provided.
+                Defaults to None, and it is set to 0.25 in _divide_dataset function if no dataset_val is provided.
+
             class_weights ([list or bool], optional): weights provided to the cross entropy loss function.
-                    The user can either input a list of weights or let DeepRanl-GNN (True) define weights
-                    based on the dataset content. Defaults to None.
+                    The user can either input a list of weights or let DeepRank-GNN (True) define weights
+                    based on the dataset content. 
+                    Defaults to None.
+
```

Comment:
the way class_weights is currently implemented is only as a boolean. It just checks whether or not it's True and then ignores what is inside.  
We should either update the documentation or (better) implement allowing custom weights.

Reply:
I changed the docstring to reflect what is happening now and added an issue (#256) to implement allowing user defined weights